controllers/openstackephemeralheat_controller.go

```diff
@@ -96,13 +96,25 @@ func (r *OpenStackEphemeralHeatReconciler) Reconcile(ctx context.Context, req ct
 		return ctrl.Result{}, err
 	}
 
-	password := k8s_rand.String(10)
 	cmLabels := common.GetLabels(instance, openstackclient.AppLabel, map[string]string{})
+	// only generate the password secret once
+	passwordSecret, _, err := common.GetSecret(r, "ephemeral-heat-"+instance.Name, instance.Namespace)
+	if err != nil && k8s_errors.IsNotFound(err) {
+		passwordSecret = openstackephemeralheat.PasswordSecret("ephemeral-heat-"+instance.Name, instance.Namespace, cmLabels, k8s_rand.String(10))
+		_, op, err := common.CreateOrUpdateSecret(r, instance, passwordSecret)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if op != controllerutil.OperationResultNone {
+			r.Log.Info(fmt.Sprintf("Secret %s successfully reconciled - operation: ephemeral-heat-%s", instance.Name, string(op)))
+		}
+	}
+
 	envVars := make(map[string]common.EnvSetter)
 	templateParameters := make(map[string]interface{})
 	templateParameters["MariaDBHost"] = "mariadb-" + instance.Name
 	templateParameters["RabbitMQHost"] = "rabbitmq-" + instance.Name
-	templateParameters["MariaDBPassword"] = password
+	templateParameters["MariaDBPassword"] = passwordSecret.Data["password"]
```

Comment:
This isn't quite right yet. Need to either convert to string and/or base64.decode?

Reply:
I think a base64 decode might do the trick

Reply:
so I tried just adding string() and it seems to work fine. You do need to base64.decode if you access it from the CLI but apparently not via the Golang libraries